torch/distributed/distributed_c10d.py

```diff
@@ -2027,6 +2027,7 @@ def gather(tensor,
         Async work handle, if async_op is set to True.
         None, if not async_op or if not part of the group
 
+    .. note:: Note that this API is not supported when using the NCCL backend.
```

Comment:
Isn't our API doc already highlighted this? https://pytorch.org/docs/stable/distributed.html. If you would like to mention this in individual APIs, please also do this for the rest of backend and collective API for consistency purposes. 

<img width="872" alt="Screen Shot 2021-04-13 at 1 22 42 PM" src="https://user-images.githubusercontent.com/16999635/114594544-5d86bb80-9c5b-11eb-89fa-2e0489cac157.png">

Reply:
agreed, we need to do a more general cleanup wrt to we document this. I was addressing user complaining about gather/gather_object inconsistency https://github.com/pytorch/pytorch/issues/55893.